crates/apps/src/lib/cli.rs

```diff
@@ -4156,6 +4156,12 @@ pub mod args {
             let tx_code_path = PathBuf::from(TX_INIT_ACCOUNT_WASM);
             let public_keys = PUBLIC_KEYS.parse(matches);
             let threshold = THRESHOLD.parse(matches);
+
+            if threshold <= Some(0) {
```

Comment:
`threshold` is optional. if it happens to be equal to `None` (i.e. it wasn't provided as a CLI arg), I think this condition will be equal to `true`, effectively making the `threshold` required, but the error message somewhat inaccurate. it comes down to the way `Ord` is implemented for `Option<T>` (`None < Some(x)` for all `x`).

even if it was `Some`, I think it would make more sense to compare it for equality with `Some(0)`
```suggestion
            if threshold == Some(0) {
```

in the end, the better solution would be to make `THRESHOLD` return an `Option<NonZeroU8>` IMO, but this requires more code changes (not many to be fair, and you just have to follow the errors returned by `make clippy`)

Reply:
I've looked at the usages of threshold in the codebase and there are around 30 usages/definitions of `threshold: u8` or `threshold: Option<u8>`

I really have no idea which one are relevant for that. This seems like the cleaner solution indeed, but can you guide me into where it's relevant to fix ?
Im open to do pair programming on that if you wish / it's worth your time.

Reply:
@sug0 Im not sure this is going the right way. I've tried since yesterday to replace most/all usage of u8 for threshold, with NonZeroU8, which leads to a lot of unwrapping/converting to compare with other number (most vec.len() ). And very time I fix one, I have 3 more errors that crop out.
Seems like some simple defensive programming at the entrypoint would work better.